packages/angular-rsbuild/src/lib/config/helpers.ts

```diff
@@ -47,7 +46,9 @@ export async function deleteOutputDir(
 
     // Leave requested directories. This allows symlinks to continue to function.
     if (directoriesToEmpty?.has(fullEntry)) {
-      await deleteOutputDir(resolvedOutputPath, fullEntry);
+      // compute its path relative to `root`, then recurse as in deleteOutputDir resolve is called on root
+      const rel = relative(root, fullEntry);
+      await deleteOutputDir(root, rel, preserveDirs);
```

Comment:
the recursive call was:
- passing a already resolved path
- not passing `preserveDirs` array